gcc -E does nothing for .s, need .S for preprocess to output

--- openssl-1.0.1j/crypto/Makefile~	2015-01-01 23:00:33.000000000 +0200
+++ openssl-1.0.1j/crypto/Makefile	2015-01-01 23:00:35.000000000 +0200
@@ -79,7 +79,7 @@
 ppccpuid.s:	ppccpuid.pl;	$(PERL) ppccpuid.pl $(PERLASM_SCHEME) $@
 pariscid.s:	pariscid.pl;	$(PERL) pariscid.pl $(PERLASM_SCHEME) $@
 alphacpuid.s:	alphacpuid.pl
-	(preproc=/tmp/$$$$.$@; trap "rm $$preproc" INT; \
+	(preproc=/tmp/$$$$.$@.S; trap "rm $$preproc" INT; \
 	$(PERL) alphacpuid.pl > $$preproc && \
 	$(CC) -E $$preproc > $@ && rm $$preproc)
 
--- openssl-1.0.1j/crypto/modes/Makefile~	2014-10-15 15:54:35.000000000 +0300
+++ openssl-1.0.1j/crypto/modes/Makefile	2015-01-01 23:17:28.000000000 +0200
@@ -53,7 +53,7 @@
 ghash-sparcv9.s:	asm/ghash-sparcv9.pl
 	$(PERL) asm/ghash-sparcv9.pl $@ $(CFLAGS)
 ghash-alpha.s:	asm/ghash-alpha.pl
-	(preproc=/tmp/$$$$.$@; trap "rm $$preproc" INT; \
+	(preproc=/tmp/$$$$.$@.S; trap "rm $$preproc" INT; \
 	$(PERL) asm/ghash-alpha.pl > $$preproc && \
 	$(CC) -E $$preproc > $@ && rm $$preproc)
 
--- openssl-1.0.1j/crypto/sha/Makefile~	2014-10-15 15:54:34.000000000 +0300
+++ openssl-1.0.1j/crypto/sha/Makefile	2015-01-01 23:19:33.000000000 +0200
@@ -60,7 +60,7 @@
 	$(PERL) $< $(PERLASM_SCHEME) $@
 
 sha1-alpha.s:	asm/sha1-alpha.pl
-	(preproc=/tmp/$$$$.$@; trap "rm $$preproc" INT; \
+	(preproc=/tmp/$$$$.$@.S; trap "rm $$preproc" INT; \
 	$(PERL) asm/sha1-alpha.pl > $$preproc && \
 	$(CC) -E $$preproc > $@ && rm $$preproc)
 
--- openssl-1.0.1j/crypto/bn/Makefile~	2015-01-01 23:21:25.000000000 +0200
+++ openssl-1.0.1j/crypto/bn/Makefile	2015-01-01 23:21:28.000000000 +0200
@@ -125,7 +125,7 @@
 ppc64-mont.s:	asm/ppc64-mont.pl;$(PERL) asm/ppc64-mont.pl $(PERLASM_SCHEME) $@
 
 alpha-mont.s:	asm/alpha-mont.pl
-	(preproc=/tmp/$$$$.$@; trap "rm $$preproc" INT; \
+	(preproc=/tmp/$$$$.$@.S; trap "rm $$preproc" INT; \
 	$(PERL) asm/alpha-mont.pl > $$preproc && \
 	$(CC) -E $$preproc > $@ && rm $$preproc)
 
