--- openssl-1.0.0/Makefile.org.orig	2010-04-10 11:05:35.900993086 +0200
+++ openssl-1.0.0/Makefile.org	2010-04-10 11:16:15.052993924 +0200
@@ -610,7 +610,7 @@
 			--release=$(VERSION) `basename $$i`") \
 			>  $(INSTALL_PREFIX)$(MANDIR)/man$$sec/$$fn.$${sec}$(MANSUFFIX); \
 		$(PERL) util/extract-names.pl < $$i | \
-			(grep -v $$filecase "^$$fn\$$"; true) | \
+			(grep -v $$filecase "^$${fn#openssl_}\$$"; true) | \
 			(grep -v "[	]"; true) | \
 			(cd $(INSTALL_PREFIX)$(MANDIR)/man$$sec/; \
 			 while read n; do \
@@ -627,11 +627,11 @@
 			--release=$(VERSION) `basename $$i`") \
 			>  $(INSTALL_PREFIX)$(MANDIR)/man$$sec/$$fn.$${sec}$(MANSUFFIX); \
 		$(PERL) util/extract-names.pl < $$i | \
-			(grep -v $$filecase "^$$fn\$$"; true) | \
+			(grep -v -E $$filecase "^$${fn#openssl_}\$$"; true) | \
 			(grep -v "[	]"; true) | \
 			(cd $(INSTALL_PREFIX)$(MANDIR)/man$$sec/; \
 			 while read n; do \
-				PLATFORM=$(PLATFORM) $$here/util/point.sh $$fn.$${sec}$(MANSUFFIX) "$$n".$${sec}$(MANSUFFIX); \
+				PLATFORM=$(PLATFORM) $$here/util/point.sh $$fn.$${sec}$(MANSUFFIX) openssl_"$$n".$${sec}$(MANSUFFIX); \
 			 done); \
 	done
 
--- openssl-1.0.0/util/point.sh.orig	2009-04-06 16:25:02.000000000 +0200
+++ openssl-1.0.0/util/point.sh	2010-04-10 11:17:19.616991410 +0200
@@ -1,7 +1,9 @@
 #!/bin/sh
 
 rm -f "$2"
-if test "$OSTYPE" = msdosdjgpp || test "x$PLATFORM" = xmingw ; then
+if true; then
+    echo ".so $1" > "$2"
+elif test "$OSTYPE" = msdosdjgpp || test "x$PLATFORM" = xmingw ; then
     cp "$1" "$2"
 else
     ln -s "$1" "$2"
