diff -Nuard openssl-0.9.7b.orig/Makefile.org openssl-0.9.7b/Makefile.org
--- openssl-0.9.7b.orig/Makefile.org	2003-04-08 13:54:32.000000000 +0200
+++ openssl-0.9.7b/Makefile.org	2003-08-20 21:12:53.000000000 +0200
@@ -578,7 +578,7 @@
 	@ ( echo 'prefix=$(INSTALLTOP)'; \
 	    echo 'exec_prefix=$${prefix}'; \
 	    echo 'libdir=$${exec_prefix}/lib'; \
-	    echo 'includedir=$${prefix}/include'; \
+	    echo 'includedir=$${prefix}/include/openssl'; \
 	    echo ''; \
 	    echo 'Name: OpenSSL'; \
 	    echo 'Description: Secure Sockets Layer and cryptography libraries and tools'; \
diff -Nuard openssl-0.9.7b.orig/Makefile.ssl openssl-0.9.7b/Makefile.ssl
--- openssl-0.9.7b.orig/Makefile.ssl	2003-04-10 22:47:14.000000000 +0200
+++ openssl-0.9.7b/Makefile.ssl	2003-08-20 21:01:09.000000000 +0200
@@ -580,7 +580,7 @@
 	@ ( echo 'prefix=$(INSTALLTOP)'; \
 	    echo 'exec_prefix=$${prefix}'; \
 	    echo 'libdir=$${exec_prefix}/lib'; \
-	    echo 'includedir=$${prefix}/include'; \
+	    echo 'includedir=$${prefix}/include/openssl'; \
 	    echo ''; \
 	    echo 'Name: OpenSSL'; \
 	    echo 'Description: Secure Sockets Layer and cryptography libraries and tools'; \
