--- openssl-0.9.6d/crypto/des/fcrypt.c.wiget	Tue Nov 20 17:37:32 2001
+++ openssl-0.9.6d/crypto/des/fcrypt.c	Tue Nov 20 17:41:40 2001
@@ -61,7 +61,7 @@
 void fcrypt_body(DES_LONG *out,des_key_schedule ks,
 	DES_LONG Eswap0, DES_LONG Eswap1);
 
-#if !defined(PERL5) && !defined(__FreeBSD__) && !defined(NeXT) && !defined(_DARWIN)
+#if !defined(PERL5) && !defined(__FreeBSD__) && !defined(NeXT) && !defined(_DARWIN) && !defined(__GLIBC__)
 char *crypt(const char *buf, const char *salt)
 	{
 	return(des_crypt(buf, salt));
--- openssl-0.9.6b/test/Makefile.ssl.wiget	Tue Nov 20 17:52:33 2001
+++ openssl-0.9.6b/test/Makefile.ssl	Tue Nov 20 17:53:46 2001
@@ -27,7 +27,7 @@
 
 DLIBCRYPTO= ../libcrypto.a
 DLIBSSL= ../libssl.a
-LIBCRYPTO= -L.. -lcrypto
+LIBCRYPTO= -L.. -lcrypto -lcrypt
 LIBSSL= -L.. -lssl
 
 BNTEST=		bntest
--- openssl-0.9.6b/crypto/des/des.h.wiget	Fri Apr 19 11:32:16 2002
+++ openssl-0.9.6b/crypto/des/des.h	Fri Apr 19 11:33:28 2002
@@ -189,7 +189,7 @@
 		  des_cblock *iv);
 char *des_fcrypt(const char *buf,const char *salt, char *ret);
 char *des_crypt(const char *buf,const char *salt);
-#if !defined(PERL5) && !defined(__FreeBSD__) && !defined(NeXT) && !defined(_UWIN)
+#if !defined(PERL5) && !defined(__FreeBSD__) && !defined(NeXT) && !defined(_UWIN) && !defined(__GLIBC__)
 char *crypt(const char *buf,const char *salt);
 #endif
 void des_ofb_encrypt(const unsigned char *in,unsigned char *out,int numbits,
