diff -urbB openssl-0.9.7h.org/crypto/evp/evp.h openssl-0.9.7h/crypto/evp/evp.h
--- openssl-0.9.7h.org/crypto/evp/evp.h	2005-05-17 21:48:42.000000000 +0200
+++ openssl-0.9.7h/crypto/evp/evp.h	2005-10-13 08:18:24.145355808 +0200
@@ -132,7 +132,11 @@
 #define EVP_CAST5_KEY_SIZE		16
 #define EVP_RC5_32_12_16_KEY_SIZE	16
 */
+#ifdef OPENSSL_FIPS
 #define EVP_MAX_MD_SIZE			64	/* longest known SHA512 */
+#else
+#define EVP_MAX_MD_SIZE		(16+20)		/* The SSLv3 md5+sha1 type */
+#endif
 #define EVP_MAX_KEY_LENGTH		32
 #define EVP_MAX_IV_LENGTH		16
 #define EVP_MAX_BLOCK_LENGTH		32
diff -urbB openssl-0.9.7h.org/crypto/hmac/hmac.h openssl-0.9.7h/crypto/hmac/hmac.h
--- openssl-0.9.7h.org/crypto/hmac/hmac.h	2005-05-07 19:21:29.000000000 +0200
+++ openssl-0.9.7h/crypto/hmac/hmac.h	2005-10-13 08:18:56.523433592 +0200
@@ -64,7 +64,11 @@
 
 #include <openssl/evp.h>
 
+#ifdef OPENSSL_FIPS
 #define HMAC_MAX_MD_CBLOCK	128
+#else
+#define HMAC_MAX_MD_CBLOCK	64
+#endif
 
 #ifdef  __cplusplus
 extern "C" {

