--- openssl-0.9.7c.pius/Makefile	Tue Sep 30 14:38:13 2003
+++ openssl-0.9.7c/Makefile	Tue Sep 30 21:29:56 2003
@@ -834,7 +834,6 @@
 		fi; \
 	fi
 	cp openssl.pc $(INSTALL_PREFIX)$(INSTALLTOP)/lib/pkgconfig
-	chmod 644 $(INSTALL_PREFIX)$(INSTALLTOP)/lib/pkgconfig
 
 install_docs:
 	@$(PERL) $(TOP)/util/mkdir-p.pl \
--- openssl-0.9.7c.pius/Makefile.org	Mon Aug 11 11:37:17 2003
+++ openssl-0.9.7c/Makefile.org	Tue Sep 30 23:13:31 2003
@@ -832,7 +832,6 @@
 		fi; \
 	fi
 	cp openssl.pc $(INSTALL_PREFIX)$(INSTALLTOP)/lib/pkgconfig
-	chmod 644 $(INSTALL_PREFIX)$(INSTALLTOP)/lib/pkgconfig
 
 install_docs:
 	@$(PERL) $(TOP)/util/mkdir-p.pl \
--- openssl-0.9.7c.pius/Makefile.ssl	Tue Sep 30 14:38:13 2003
+++ openssl-0.9.7c/Makefile.ssl	Tue Sep 30 21:29:56 2003
@@ -834,7 +834,6 @@
 		fi; \
 	fi
 	cp openssl.pc $(INSTALL_PREFIX)$(INSTALLTOP)/lib/pkgconfig
-	chmod 644 $(INSTALL_PREFIX)$(INSTALLTOP)/lib/pkgconfig
 
 install_docs:
 	@$(PERL) $(TOP)/util/mkdir-p.pl \
