--- openssl-0.9.5/apps/CA.pl.wiget	Mon Feb 28 15:30:31 2000
+++ openssl-0.9.5/apps/CA.pl	Fri Mar  3 10:09:56 2000
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 #
 # CA - wrapper around ca to make it easier to use ... basically ca requires
 #      some setup stuff to be done before you can use it and this makes
--- openssl-0.9.5/apps/der_chop.in.wiget	Wed May 26 00:31:49 1999
+++ openssl-0.9.5/apps/der_chop.in	Fri Mar  3 10:10:29 2000
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 #
 # der_chop ... this is one total hack that Eric is really not proud of
 #              so don't look at it and don't ask for support
