--- openssl-0.9.7.old/crypto/Makefile.ssl	Fri Dec 20 15:28:45 2002
+++ openssl-0.9.7/crypto/Makefile.ssl	Sun Feb 16 20:02:43 2003
@@ -50,7 +50,7 @@
 top:
 	@(cd ..; $(MAKE) DIRS=$(DIR) all)
 
-all: buildinf.h lib subdirs shared
+all: shared
 
 buildinf.h: ../Makefile.ssl
 	( echo "#ifndef MK1MF_BUILD"; \
@@ -96,7 +96,7 @@
 	$(RANLIB) $(LIB) || echo Never mind.
 	@touch lib
 
-shared:
+shared: buildinf.h lib subdirs
 	if [ -n "$(SHARED_LIBS)" ]; then \
 		(cd ..; $(MAKE) $(SHARED_LIB)); \
 	fi
--- openssl-0.9.7.old/ssl/Makefile.ssl	Thu Dec 19 22:10:20 2002
+++ openssl-0.9.7/ssl/Makefile.ssl	Sun Feb 16 20:03:13 2003
@@ -55,14 +55,14 @@
 top:
 	(cd ..; $(MAKE) DIRS=$(DIR) all)
 
-all:	lib shared
+all:	shared
 
 lib:	$(LIBOBJ)
 	$(AR) $(LIB) $(LIBOBJ)
 	$(RANLIB) $(LIB) || echo Never mind.
 	@touch lib
 
-shared:
+shared: lib
 	if [ -n "$(SHARED_LIBS)" ]; then \
 		(cd ..; $(MAKE) $(SHARED_LIB)); \
 	fi
