diff -ruN openssl-1.0.2.orig/crypto/sha/sha.h openssl-1.0.2/crypto/sha/sha.h
--- openssl-1.0.2.orig/crypto/sha/sha.h	2015-01-22 15:58:32.000000000 +0100
+++ openssl-1.0.2/crypto/sha/sha.h	2015-01-27 16:12:51.744113627 +0100
@@ -59,8 +59,8 @@
 #ifndef HEADER_SHA_H
 # define HEADER_SHA_H
 
-# include <openssl/e_os2.h>
 # include <stddef.h>
+# include <openssl/e_os2.h>
 
 #ifdef  __cplusplus
 extern "C" {
